src/parachain/vaults.ts

```diff
@@ -950,4 +959,31 @@ export class DefaultVaultsAPI implements VaultsAPI {
         const tx = this.api.tx.vaultRegistry.acceptNewIssues(currencyPair, acceptNewIssues);
         await this.transactionAPI.sendLogged(tx, this.api.events.system.ExtrinsicSuccess, true);
     }
+
+    async getExchangeRateForLiquidation(
+        vaultAccountId: AccountId,
+        collateralCurrency: CollateralCurrencyExt
+    ): Promise<Big | undefined> {
+        const [vaultExt, liquidationRateThreshold, lockedCollateral] = await Promise.all([
+            this.get(vaultAccountId, collateralCurrency),
+            this.getLiquidationCollateralThreshold(collateralCurrency),
+            this.getLockedCollateral(vaultAccountId, collateralCurrency),
+        ]);
+
+        if (liquidationRateThreshold.eq(0)) {
+            return undefined;
+        }
+
+        const issuedTokens = vaultExt.getBackedTokens();
+
+        if (issuedTokens.toBig().eq(0)) {
+            return undefined;
+        }
+
+        // calculate the theoretical exchange rate at which the vault would be (close to) liquidated
+        const liquidationRateCollaterallPerWrapped = lockedCollateral
+            .toBig()
+            .div(issuedTokens.toBig().mul(liquidationRateThreshold));
```

Comment:
So the calculation we do on the parachain is `collateral / exchange_rate / threshold = wrapped`.

Plugging in an example to remind myself:
- exchange rate is 2333 DOT per BTC (converted to planck / sat this is `2333 * (10**10 / 10**8)`)
- if we lock 2333 DOT we can mint `(2333 * 10**10) / 233300 / 2.6 = 0.38 BTC`
- or put another way, to mint 1 BTC we need `(1 * 10**8) * 233300 * 2.6 = 6065.8 DOT`
- so to get the exchange rate for liquidation we can do `(6065.8 * 10**10) / (1 * 10**8 * 2.6) = 233300`
- or for liquidation `(6065.8 * 10**10) / (1 * 10**8 * 1.1) = ~551436.36`
- so the liquidation exchange rate in this case is 5514.36 DOT per BTC (`551436.36 / (10**10 / 10**8)`)

Therefore the calculation we need to do is `exchange_rate = collateral / (wrapped * threshold)`.

Reply:
tl;dr this implementation looks fine

Reply:
```suggestion
        const liquidationRateCollateralPerWrapped = lockedCollateral
            .toBig()
            .div(issuedTokens.toBig().mul(liquidationRateThreshold));
```